src/main/java/com/ratifire/devrate/util/zoom/webhook/service/ZoomWebhookService.java

```diff
@@ -36,17 +36,21 @@ public String handleZoomWebhook(String requestBody, Map<String, String> headers)
 
     String signature = headers.get("x-zm-signature");
     String timestamp = headers.get("x-zm-request-timestamp");
-
     zoomWebhookAuthService.validateSignature(signature, requestBody, timestamp);
 
     WebHookRequest payload = objectMapper.readValue(requestBody, WebHookRequest.class);
     String event = payload.getEvent();
 
-    return switch (event) {
-      case "endpoint.url_validation" -> zoomWebhookAuthService.handleUrlValidationEvent(payload);
-      case "meeting.ended" ->
+    switch (event) {
+      case "endpoint.url_validation":
+        return zoomWebhookAuthService.handleUrlValidationEvent(payload);
+      case "meeting.ended":
+        if (interviewCompletionService.validateMeetingEndTime(payload)) {
```

Comment:
I'm not sure, but maybe it's better to move this conditioning to the method `completeInterviewProcess` and leave here only 
`return interviewCompletionService.completeInterviewProcess(payload.getPayload().getMeeting());`

**Let's someone else watch and comment this proposal**

Reply:
Agree with Ivan. imho make sense